ngraph/test/runtime/interpreter/evaluates_map.cpp

```diff
@@ -214,6 +215,55 @@ namespace
         return true;
     }
 
+    namespace bin_conv_v1
+    {
+        template <element::Type_t t_in, element::Type_t t_f>
+        inline void evaluate(const shared_ptr<op::v1::BinaryConvolution>& op,
+                             const HostTensorVector& outputs,
+                             const HostTensorVector& inputs)
+        {
+            using T_IN = typename element_type_traits<t_in>::value_type;
+            using T_F = typename element_type_traits<t_f>::value_type;
+
+            const auto in_data_ptr = inputs[0]->get_data_ptr<T_IN>();
+            const auto filter_data_ptr = inputs[1]->get_data_ptr<T_F>();
+            auto out_data_ptr = outputs[0]->get_data_ptr<T_IN>();
+            const auto in_shape = inputs[0]->get_shape();
+            const auto filter_shape = inputs[1]->get_shape();
+            const auto out_shape = outputs[0]->get_shape();
+
+            runtime::reference::binary_convolution<T_IN, T_F>(in_data_ptr,
+                                                              filter_data_ptr,
+                                                              out_data_ptr,
+                                                              in_shape,
+                                                              filter_shape,
+                                                              out_shape,
+                                                              op->get_strides(),
+                                                              op->get_dilations(),
+                                                              op->get_pads_begin(),
+                                                              op->get_pads_end(),
+                                                              op->get_pad_value());
+        }
+    } // bin_conv_v1
+
+    template <element::Type_t ET>
+    bool evaluate(const shared_ptr<op::v1::BinaryConvolution>& op,
+                  const HostTensorVector& outputs,
+                  const HostTensorVector& inputs)
+    {
+        switch (inputs[1]->get_element_type())
+        {
+        case element::Type_t::u1:
+            bin_conv_v1::evaluate<ET, element::Type_t::u8>(op, outputs, inputs);
+            break;
+        default:
+            throw std::runtime_error(
+                "BinaryConvolution supports only u1 element type for filters input");
```

Comment:
Why can't we do this type check in  runtime::reference::binary_convolution?

Reply:
There is no reason to call runtime::reference::binary_convolution if filters are not u1 type. That is why I decided to thow runtime exception in this case.